% -*- root: warwickthesis.tex -*-

\section{Biological background}
\label{sec:biol-backgr}

In biological systems, distinct states and especially states that are indicative of disease can often be characterised by changes in gene expression levels \citep{DeRisi:1997dw,Spellman:1998wj,Eisen:1999uw, Brown:1999bk}. It is important to first understand the role played by genes in cells and also what it is we mean by expression of a gene, therefore below we present a brief introduction to these topics as well as specific examples where changes in gene expression have been found to play a central role in changes of cell states.

\subsection{The cell}
\label{sec:cell}

If we define reproduction as a basic principle of life, the cell is the smallest unit that autonomously allows for this process. Independent of the process of reproduction, to ensure a faithful reproduction of organisms, information is passed on from one generation to the next. This information takes the form of a molecule, Deoxyribonucleic acid (DNA), organised in a double helix structure made up of two DNA strands. Information on such a strand is stored as a sequence of four distinguishable subunits. These four subunits are adenine (A), guanine (G), cytosine (C) and thymine (T), where C-G and A-T base pairs held together by hydrogen bonds make up the DNA double helix. Some sequences of the DNA code for proteins (coding region) and are known as genes.  There is still debate about the extent to which DNA is made up of coding regions and noncoding regions; the difference between organisms can be very large. In humans only about $2\%$ of DNA is considered to specifically contain genes \citep{Elgar:2008dm}. The initial idea that a major part of noncoding DNA is junk has been refuted recently as part of the international project of the ENCODE consortium \citep{Pennisi:2012wl}. Some of these areas have been shown to contain areas for proteins to bind and influence gene activity, or stretches where chemical modifications can  switch off part of the DNA. It has also been established that regulation of genes is much more involved than previously thought and areas of DNA far away from a given position can influence gene expression at that position. The first step of converting coding regions of the DNA to proteins is known as transcription. In this process, information from the DNA is read off and converted into a single stranded molecule of ribonucleic acid (RNA); more specifically the type of RNA used is known as messenger RNA (mRNA). The following step involves some post-transcriptional modifications, where the mRNA is modified to mature mRNA. There are many such modifications the cell performs, but an important one is splicing. This process removes regions of mRNA called introns\footnote{a region inside a gene sequence that bind together the gene and are subsequently removed} that do not code for proteins after which the remaining exons are \emph{spliced} together. There are multiple ways to splice a set of exons which can lead to many different proteins being read from the same sequence on DNA. Finally proteins are synthesised from mature mRNA during the process known as translation. During translation mRNA information is read in groups of three subunits called codons, hence there are $4^3 = 64$ codons which are mapped to $20$ amino acids used to make proteins. These final DNA products then perform numerous functions in the cell including steps in their own synthesis. These three steps together are known as gene expression and form the central dogma of molecular biology, for a Summary see Figure \ref{fig:gene-expression}. The DNA molecules contained in cells is more than two metres long and to economically pack them inside cells, which also contain vast amounts of other material, they are normally in packed structures known as chromatin. This is quite simply a combination of the DNA proteins which allow DNA be wrapped and packed very tightly in the nucleus of the cell. In addition to packing DNA, it also serves a functional purpose in that the topology of the packing hides and exposes certain genes on DNA. In a system where gene expression occurs via several protein dependant steps hidden genes cannot be expressed since regulatory proteins cannot interact with them.

\begin{figure}[!t]
  \centering
  \includegraphics[width=0.69\textwidth]{pics/dogma-bio.pdf}
  \caption{Gene expression. The central dogma of molecular biology states that genes on DNA are transcribed to mRNA which is transformed to mature mRNA by post-transcriptional modifications such as splicing. The mature mRNA is then translated to proteins. This final product which plays a central role in the functions of cells in turn also influences the transcription step as well as the post-transcriptional modification step.}
  \label{fig:gene-expression}
\end{figure}

Each step during gene expression outlined above happens at a different time scale therefore when interpreting biological data it is important to keep this in mind; the effect one sees on gene expression at a given time might be due to a protein interaction process which took place at an earlier time point. The time scale can vary from seconds for proteins \citep{Herce:kq} up to $16$ hours for the largest gene \citep{Tennyson:1995dl}. This of course becomes even more complicated once interactions between the different players are accounted for.

The control of gene expression is responsible for characteristics of a cell. As indeed a lung cell and a brain cell have  identical genetic information, but gene expression is responsible for defining a cells purpose. This is in part controlled by material outside of the coding region. Daughter cells in addition to inheriting genetic information from parent cells, also inherit information that determines the characteristics of the cell unrelated to DNA, i.e. daughter cells of a cell in lungs will remain lung cells. The umbrella term often used to describe all types of material that could pass on this information not directly part of DNA is \emph{epigenetic material}. The final word on epigenetics has yet to be spoken but two types of information passed on in this manner are: Firstly, the chromatin structure which determines active and inactive genes and to some extent the way they are read off. Active sections of chromatin are called euchromatin and inactive parts are called heterochromatin. Secondly, DNA methylation which is the addition of a methyl group to certain bases in DNA, altering expression of genes.

Biological changes in state, which are the focus of this work, can be described in many different ways. Some changes in state are due to changes in gene expression. Other changes in state can also be epigenetic without direct changes in gene expression. Although both of these types of state changes are deeply interlinked they can occur at vastly different time scales. Here we will focus on state changes directly due to changes in gene expression.

Further information on the cell in general as well as epigenetics can be found in \citet[Chapters~1,7]{Alberts:2007tv}. A specific overview of epigenetics has been attempted in \cite{Goldberg:2007tl} and its effect on gene expression in \cite{Gibney:2010ws}.

\subsection{Cancer biology}
\label{sec:cancer-biology}

An area of biology that is of great importance due to its impact on large populations and where state changes play a critical role is the study of cancer. Cancer is an extremely complex disease and attempts have been made to determine basic underlying principles, which the authors refer to as the 'Hallmarks of cancer' \citep{Hannah:2000wo, Hanahan:2011gu}. Genetic aberrations play a central role in this disease as many carcinogens (agents that can cause cancers) directly influence DNA sequences or are themselves mutations of genes naturally occurring in the cell. These defects range from point mutations on single base pairs all the way to deletions of large sections of DNA. Especially the process of cell division is highly susceptible to such attacks; in healthy cells there is a DNA repair mechanism in place that prevents  changes from becoming permanent or leads to apoptosis (programmed cell death) if repair is impossible.

One important principle shared among cancer types is the unbound proliferation of cells leading to the build up of concentrated cell masses, also known as tumours. Note that many such tumours are benign since they do not transform into cancers. The bodies defence mechanisms against such unchecked proliferation are circumvented either by introduction of oncogenes or mutations in tumour suppressor genes. Oncogenes are mutations in genes that result in a protein which drives uncontrolled cellular proliferation, resulting in a tumour. These proteins do not respond to the natural signals that inhibit cell division, hence the proliferation is uncontrolled and cannot be kept in check by the cells defence mechanisms against such growth. Alternatively mutations can occur in genes responsible for onset of apoptosis or DNA repair (also known as anti-oncogenes). Genetic mutations are especially problematic if they occur in the germline\footnote{cells from which egg or sperm cells are derived}, as exemplified by Liâ€“Fraumeni syndrome \citep{Li:1969kl}, where the mutation in an essential tumour suppressor gene is passed on to offspring and leads to hereditary predisposes to a large number of cancer types.

It is important to note that one mutation or defect is not sufficient to lead to the development of cancers; in fact several processes need to be affected. Additional processes developed by cells to defend against invading cancers include a limit on the number of times a cell can divide. Cells also rely on external growth signals to start division as well as external growth inhibition signals to stop division. Cancers are known for an uninhibited expansion, for which they need nutrients and therefore develop the ability to initiate the deployment of additional blood vessels; a process known as angiogenesis. Tumorous cell become especially dangerous if they develop the ability to brake off from the main tumour and invade surrounding organs, tissues or even distant parts of the body; this known as metastasis.

As mentioned above an oncogene is a gene that can cause cancer and the first such oncogene (v-Src) was discovered in the late 1970s and early 1980s \cite{Martin:2001jx} after the initial discovery almost 70 years earlier; hinting at the possibility to induce solid tumours in chicken using a filtered agent by \cite{Rous:1911uf}. Investigating the Rous sarcoma virus in chicken the v-Src was discovered. Further research determined that a variant of v-Src, called c-Src is also contained in normal chicken. This discovery  fundamentally changed the understanding of cancer which until then had been ascribed to  viral causes. Later this gene was also found in humans and since this discovery it is probably the most widely studied oncogene; despite this there remain many unknowns. The protein from this oncogene has many downstream interactions with numerous other proteins. Hence it is not surprising that in almost $50\%$ of tumours originating in breast, colon, liver, lung and the pancreas the c-Src interaction pathway is activated \cite{Dehm:2013fr}. Due to mutations, c-Src is overexpressed\footnote{expression is higher than normal} and activated leading to the constant activation of downstream signalling pathways that ensure survival, proliferation and invasion, an therefore to development of cancers.

Further details on cancer biology and the role of genes can be found in \cite{Weinberg:2013uu}.

\subsection{Stem cells}
\label{sec:stem-cells}

Central to cellular development is the creation of distinct differentiated cell types from a small collection of undifferentiated cells in the embryo referred to as embryonic stem cells. Clearly this transformation involves state changes on the genetic and epigenetic level. In recent years there has been an increase in research in these areas especially due to its potential in applications to personalised medicine; the next big frontier in medicine. The idea is to enable induction of alternative cell fates from embryonic cells or to enable development of cells that allow for a change in cell fates of tissues or blood samples. Examples include the development of neurons from cells that are responsible for creating extra cellular matrix known as fibroblasts \citep{Vierbuchen:2010fa, Pang:2011ce} or the development of muscle cells from fibroblasts \citep{Ieda:2010ir, Efe:2011bpa}. All types of undifferentiated cells that can produce differentiated cell types are comprehensively known as stem cells (SC). Another property shared by all stem is cells that they can differentiate to produce more stem cells multiple times. Broadly speaking there are two types of stem cells. Embryonic stem cells (ES cells) are cells derived from an embryo in its early development and  adult (or somatic) stem cells which are found in fully developed organs. The main difference between the two types is how many types of cells they can differentiate into. ES cells are pluripotent i.e. they can differentiate into many possible cell types. Adult SCs, also known as multi-potent stem cells, can only differentiate into limited cell types often serving the function of replenishing damaged cells of a single organ. For medical applications of course ES cells are more useful, but for some people there are ethical concerns associated with their usage and their harvesting; independently of the rationale behind these concerns, it does create issues in research if such cells are used.

A new approach, proposed by \cite{Takahashi:2006hi}, is to use differentiated somatic cells and derive induced pluripotent stem cells (iPS cells) which have distinct advantages if successful. These iPS cells are able to differentiate to various cell types and could in future allow for personalised medicine. The process in creating iPS cells involves artificially inducing $4$ genes (reprogramming factors) for several days and they do indeed find cells which have properties comparable to ES cells. More detailed studies show that iPS cells are influenced by the used reprogramming factors and there are epigenetic differences between ES cells and iPS cells \citep{Carey:2011bs, Bock:2011kx}. One important concern is the difference in DNA methylation of iPS cells and ES cells in terms of epigenetic material that would make their use difficult, this problem is now being addressed \citep{Bagci:2013ey} as well as other safety concerns.

Further information on stem cells, ES cells, somatic cells as well as iPS cells, can be found in \cite{lanza2009essentials} and in \cite{Lanza:2013uk}.

\subsection{Cell cycle}
\label{sec:cell-cycle}

An essential step governing the division, differentiation and maturation of all cells is the cell cycle. In simple terms it is the process by which two daughter cells are produced from one mother cell by duplication of the cell contents; most importantly the DNA. Details of the process can vary between organisms as well as between different stages of development. Most cells in the human body are not taking part in the cell cycle, but are in a resting phase. The most basic principles of the mammalian cell cycle can be summarised into four phases:

\begin{itemize}
\item {\bf G$_1$ phase} The first gap phase during which cells increase in size. It also includes a restriction point up to which the cell is driven by external stimuli and after which it can progress through G$_1$ independently.
\item {\bf S phase} The transition from the G$_1$ to the S (Synthesis) phase contains a powerful checkpoint after which the cell is committed to duplication. During the S phase itself DNA is replicated.
\item {\bf G$_2$ phase} The second gap phase is not present in all organisms. In short the cell keeps increasing in size, synthesises proteins and prepares for mitosis. It contains a checkpoint to determine DNA damage and stops the process.
\item {\bf M phase} The final step in the cell cycle is the mitotic (M) phase. The duplicated chromosomes are separated into two cells and a new nucleus is created. The M phase also contains a checkpoint to ensure the cell is ready for division.
\end{itemize}

Despite all these checks and balances in place during the cell cycle, uncontrolled cell division  still occurs in tumourigenesis as mentioned above. Intervention on the cell cycle plays a central role in unbound growth of cells. In many cases proteins essential during check points are mutated, inhibited or overexpressed \citep{Williams:2012eg}. Understanding and perturbing elements in the cell cycle could be a good approach for potential cancer treatments since the mammalian cell cycle is conserved across a variety of cell types; at the same time it plays a central role in cancers.

The cell cycle is also sensitive to UV radiation which is a well known carcinogen. UV radiation incident on a cell can lead to damage to DNA, and if the damage is too extensive, cells can undergo apoptosis. If the damage is not too large some cells will arrest and re-enter the cell cycle at a later time. Radiation has an effect on gene expression which is also related to the intensity of the radiation as different genes become active to respond to the stimulus, but more importantly the type of response is driven by expression of certain genes. \citep{Gentile:2003in}.

More information about the cell cycle can be found in \citet[Chapter~17]{Alberts:2007tv} and its relationship to cancer in \citet[Chapter~8]{Weinberg:2013uu}.


\section{Experimental background}
\label{sec:exper-backgr}

In this section we explore two techniques to obtaining time-course assays for genome wide gene expression data. Current techniques for such measurement are only possible on a population level. There exist techniques for single cell measurements \citep{Buganim:2012hp}, but these techniques are not yet fully developed and only allow measurement of a limited number of genes. Initially to be able to obtain these measurements is to create a homogenate from the sample which is done by mechanically breaking down cells using a variety of different procedures.  After which different subsets are filtered out of the mixture to perform microarray experiments.  For RNA sequencing (RNA-seq) mRNA are filtered out.

\subsection{Microarray}
\label{sec:microarray}

Different stages of gene expression can be measured using microarrays. Here we present the one most commonly used; the DNA microarray with the aim is to measure mRNA levels. There are two types of DNA microarray, the cDNA \citep{Hughes:2001ho} and the high density oligonucleotide chips \citep{Lockhart:1996jw}. Here we discuss cDNA chips as they are relevant to this work. The basic principle of cDNA\footnote{Complementary DNA, DNA reverse-transcribed from mRNA} microarrays is based on high-density array with DNA sequences printed on them. The sample mRNA reverse-transcribed to cDNA labelled in two different colours (red and green). Equal proportions of the labelled colours are mixed together and hybridised to the array and using a scanner fluorescence measurements are made for each colour. The resulting expression is obtained by the ratio of the measurements in each colour, see \cite{phimister1999chipping} for further information. To ensure measurements are comparable between different samples and even different experiments it is important to normalise each sample. The most robust normalisation method is to based on subtracting a position and intensity ($A$) dependant constant from the log ratios of the intensity measurements in green $G$ and red $R$, where the colours are achieved by staining the two cDNA populations:

\begin{equation}
  \label{eq:microarray-norm}
  \log_2 \frac{R}{G} - l(A, j),
\end{equation}
where $l(A, j)$ is the lowess fit \citep{Cleveland:2012fu} to the plot of $\log_2 R$ against $\log_2 G$ rotated counterclockwise by $45^{\circ}$. More detail on normalisation and cDNA microarray measurements can also be found in \cite{Dudoit:2002va}.

\subsection{RNA sequencing}
\label{sec:rna-sequencing}

Until recently the most prevalent method for obtaining gene expression data which is essential to understanding disease states has been DNA microarray measurement. One drawback is that observations are relative and indirect i.e. measurements via fluorescence intensity and ratio of two colours. This approach of using two colour is especially useful for normalising measurement. Additionally microarray measurements are limited by prior knowledge of genes, since arrays can only be constructed to include sequences of known genes. A new contender that attempts to address some of the shortcomings of these is RNA-seq developed roughly $5$ years ago \citep{Mortazavi:2008jj, Nagalakshmi:2008cj}. RNA-seq measurements are integer count data and cover the whole genome.

\subsubsection{Experimental protocol}
\label{sec:technique-bio}

RNA-seq experiments also attempt to measure mRNA obtained from homogenate. The first step is a random fragmentation of the sample mRNA. The next step is a reverse-transcription of the fragmented samples to cDNA. Next comes a polymerase chain reaction (PCR) step which is a method of amplification to obtain more copies of DNA from few copies of a slice of DNA and is performed multiple times. This step is the source of one type of systematic error since different sections of  DNA have a different susceptibility to a PCR amplification. In the next step each fragment is sequenced in a high throughput machine; resulting sequences are referred to as \emph{reads}. These reads can now be mapped to a known genome or transcriptome\footnote{collection of all RNA} resulting for one count for each fragment of gene found.  Alternatively reads can also be used to construct a transcriptome without mapping it to a known genome (de novo assembly). Figure \ref{fig:li-biostats} from \cite{Li:2012ea} summarises this protocol.

\begin{figure}
  \centering
  \includegraphics[width=0.9\textwidth]{pics/li-biostats12.jpg}
  \caption{RNA-sequencing (RNA-seq). The most commonly used modern genome wide assay. From a homogenate cell sample mRNA is filtered out and passes through this pipeline to give integer count expression values for genes. Figure from Li et al., Normalization, testing, and false discovery rate estimation for RNA-sequencing data, Biostatistics, 2012, 13(3), by permission of Oxford University Press.}
  \label{fig:li-biostats}
\end{figure}

\subsubsection{Normalisation}
\label{sec:normalisation}

Though RNA-seq avoids some of the issues associated with microarray measurements it still has it's own difficulties that need to be overcome before analysing the data. The first as, already mentioned above, is the systematic error due to the PCR step. Another is due to the random fragmentation step; since larger genes will have more fragments resulting in a higher per gene count. Due to these reasons the total number of counts is also not conserved across different samples.

Therefore there is a real need for a normalisation step prior to comparison of data, to ensure the effect of these problems are minimised. The issue of different gene lengths can be removed by simply normalising for gene length which is a known quantity when  mapping counts to a genome. Additionally this is only an issue when comparing genes in the same sample. For comparisons between samples this step is unnecessary since gene length is constant across samples. The question of normalisation between samples especially arises in differential expression. In RNA-seq data, unlike for microarray data, the question of normalisation has yet to be settled. The normalisation method most commonly used is to normalise all samples to a fixed number of reads, but this leads to issues as can be illustrated using a simple example.

Say $N_{ij}$ is the total number of reads in experiment $i$ for gene $j$. In a simple case $N_{1j} \simeq 2 \, N_{2j}$ for all genes and hence sequencing depth of experiment $1$ is twice the sequencing depth of experiment $2$. In a slightly more complex example the issue becomes clear. If we now consider $j = 1, \ldots,  100$ to have $N_{1j} = 1$ and $N_{2j} = 2$ and for $j = 101$ $N_{1j} = 100$ and $N_{2j} = 0$ both experiments have the same sequencing depth which would suggest all genes are differentially expressed. But it would be more realistic to consider that the sequencing depth of experiment $2$ is twice that of experiment $1$ and that only gene $j = 101$ is differentially expressed. Therefore a good strategy would be identify genes that are not differentially expressed and calculate a sequencing depth just for those and use it for the whole sample. One idea would be to identify housekeeping genes\footnote{genes that are supposed to remain constant} for the particular application at hand and calculate a sequencing depth, but this is an unsatisfactory solution since it does not generalise (each application would have different housekeeping genes) and requires further experiments or prior knowledge. \cite{Bullard:2010go} propose a method using quantiles instead of total counts. \cite{Robinson:2010dd} propose a method based on total count normalisation called trimmed mean of M values (TMM). The idea is to impose a cutoff on log-fold change $M$-value and the absolute expression level $A$-value and calculate the sequencing depth using what remains. The $M$-value  and $A$-value is calculated using the definition $N_i = \sum_j N_{ij} $ as follows:

\begin{eqnarray}
  \label{eq:MA-values}
  M_j &= \log_2 \frac{N_{ij} / N_i}{N_{i'j}/ N_{i'}} \\
  A_j &= \frac{1}{2} \log_2 \left(N_{ij}/ N_i \bullet N_{i'j} / N_{i'} \right).
\end{eqnarray}

In the R package \texttt{edgeR} this method as well as a few others have been implemented for application to RNA-seq data \citep{Robinson:2010cw}.
A summary comparing different normalisation methods can be found in \cite{Oshlack:2010kr}.

%%% Local Variables:
%%% TeX-master: "warwickthesis"
%%% End:

