\chapter{Discussion and Outlook}
\label{cha:discussion-outlook}

Advancements in measuring population level transcription, proteomic abundance and epigenetic information cheaply has lead to widespread application of these methods. Additional measurements on a single-cell level have also been made in more recent years \citep{Wheeler:2003ka, Dalerba:2011cc, Wang:2010ew} but this technique has its own shortcomings but they suffer from a limitation on the amount of information that can be obtained through such techniques. For transforming biological systems single-cell measurements would be ideal even if there is a limit on the amount of information available, but current techniques require destruction of cells for a measurement of more than a handful of genes or proteins. Therefore techniques that attempt to deconvolve information on the single-cell level from population level data are extremely important for a better understanding of causes and triggers of transformation. 

Chapter \ref{cha:stamm} is focused on State transitions using aggregated Markov models (STAMM) where we expanded on previous work \citep{Armond:2013} and presented a full description of the model including assumptions. We investigated properties of the model such as (empirical) identifiability, behaviour when assumptions are broken using single-cell simulations and proposed a computationally efficient unbiased approach to parameter estimation and model selection. We showed empirically that the model is identifiable given assumptions. Under breaking model assumptions estimations are stable but if model assumptions are strongly violated state specific expression are still recovered, but transition rates can be badly estimated. Therefore estimation results should always be considered as motivation for further experiments to verify results.

In Chapter \ref{cha:oncog-transf} we applied STAMM to RNA-seq data for obtained by {\it in vitro} experiments for oncogenic transformation of a MCF-10A cell line. The system is based on of the most well studied oncogene still there remain open questions about exact steps required for the transformation. Due to a mycoplasm infection in  cells used for the {\it in vitro} study, biological conclusions need to interpreted with caution. Therefore this application is useful to illustrate that STAMM can also be applied to RNA-seq data with useful results. In future this oncogenic transformation due to its rapid transformation and clean induction can be used to validate the model in more detail and will also enable the extension of the model. Finding unique surface markers combinations for states it will be possible to identify states of cells and to isolate them and verify parameters. We also showed that this implementation of the model is computationally efficient and it takes less than $4$ hours to estimate $p=2809$ genes on fifty cores. 

% ================================================================================

Chapter \ref{cha:stem-cells} contains a quick outline of application of STAMM to a microarray time-course obtained from reprogramming differentiated cells to iPS cells. We briefly sketched the procedure used in \cite{Armond:2013} and parameters obtained during estimation. To test if estimation results are accidental or if data indeed has underlying structure we also applied STAMM to randomly permuted data. Results show that there was indeed structure in the data.  Then we compare model predictions from STAMM to recent single-cell data from a different secondary MEF experiment \citep{Buganim:2012hp}. We found that results are consistent with findings from single-cell measurements in terms of the number of intermediate states if we compare single-cells measured at different time points can be mapped well to corresponding state specific expression signatures.

In Chapter \ref{cha:cell-cycle} we proposed a model for an initially heterogeneous population where it is possible to observe ultimate cell-fate due to a stimulus. The gene expression is only measured for the initial cell population and subsequent cell-fate is measured for different stimulus strengths. Starting from basic principles we set out a description of the biological system followed by an outline of the estimation procedure based on data obtained experimentally. We set out a single cell simulation procedure for this system. We use a single gene simulation and a four gene simulation to test the model. We show that in single gene simulation it is possible to pick out parameters at a variety of noise levels. In the four gene simulation parameters are picked out well at low noise levels. At higher noise levels this is more problematic and some parameters are not estimated well. 

%%% Local Variables:
%%% TeX-master: "warwickthesis"
%%% End:
